packages/history-utility/src/history-utility.ts

```diff
@@ -38,6 +39,24 @@ export type History<T> = {
   index: number;
 };
 
+type SubscribeOps = Parameters<Parameters<typeof subscribe>[1]>[0];
+type HistorySubscribeCallback = (
+  ops: SubscribeOps,
+  historySaved: boolean
+) => void;
+
+export type HistoryOptions = {
+  /**
+   * determines if the internal subscribe behaviour should be skipped.
+   */
+  skipSubscribe?: boolean;
+
+  /**
+   * a callback function executed on each subscribe call
+   */
+  subscribeCallback?: HistorySubscribeCallback;
```

Comment:
should we call this `onHistorySaved` and only call it when the history is stored?

cc: @dai-shi

Reply:
scrap that..actually want the callback to be triggered everytime